src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -132,24 +131,30 @@ public override void Initialize()
                 }
 
                 var stopwatch = Stopwatch.StartNew();
-                MonitorServersAsync().ConfigureAwait(false);
-                // We lock here even though AddServer locks. Monitors
-                // are re-entrant such that this won't cause problems,
-                // but could prevent issues of conflicting reports
-                // from servers that are quick to respond.
-                var clusterDescription = Description.WithType(Settings.ConnectionMode.ToClusterType());
+
                 var newServers = new List<IClusterableServer>();
-                lock (_serversLock)
+                lock (_updateClusterDescriptionLock)
```

Comment:
Strictly speaking we could omit this lock since during Initialize nothing else will call UpdateClusterDescription, but it's easier in the long run to have the lock in than to document in the code why it was omitted.